xbmc/filesystem/test/TestFile.cpp

```diff
@@ -103,7 +103,7 @@ TEST(TestFile, Write)
   char buf[30];
   memset(&buf, 0, sizeof(buf));
 
-  ASSERT_TRUE((file = XBMC_CREATETEMPFILE("")) != NULL);
+  ASSERT_NE(nullptr, file = XBMC_CREATETEMPFILE(""));
```

Comment:
nullptr is the typesafe c++11 replacement of NULL, it can't accidentally be assigned to an int or anything like it. Not sure about the macro, I left it as is.